client/menus/misc/teleport.lua

```diff
@@ -34,8 +34,15 @@ local function createLocationsMenu()
         end,
         options = menuOptions
     }, function(_, _, args)
-        SetEntityCoords(cache.ped, args.coords.x, args.coords.y, args.coords.z, true, false, false, false)
-        SetEntityHeading(cache.ped, args.heading)
+        local currentVeh = GetVehiclePedIsIn(cache.ped, false)
```

Comment:
Use cache.vehicle instead of the native. It's a boolean or a vehicle id, so it's easier to check in the if statement.